modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/field/expression/handler/MembershipsUserFieldExpressionHandler.java

```diff
@@ -45,38 +45,48 @@ public class MembershipsUserFieldExpressionHandler
 	public void bindProcessorContext(
 		UserProcessorContext userProcessorContext) {
 
-		List<Long> userGroupIds = new ArrayList<>();
-
-		UserProcessorContext.UserBind<User> userBind =
-			userProcessorContext.bind(
-				_processingIndex,
-				(currentUser, newUser, serviceContext) -> {
-					_userGroupLocalService.setUserUserGroups(
-						newUser.getUserId(),
-						ArrayUtil.toArray(userGroupIds.toArray(new Long[0])));
-
-					return newUser;
-				});
-
-		userBind.mapStringArray(
-			"userGroups",
-			(user, values) -> {
-				if (values == null) {
-					return;
-				}
-
-				for (String value : values) {
-					UserGroup userGroup = _userGroupLocalService.fetchUserGroup(
-						user.getCompanyId(), value);
-
-					if (userGroup != null) {
-						userGroupIds.add(userGroup.getUserGroupId());
+		for (String validFieldExpression : getValidFieldExpressions()) {
+			if (!userProcessorContext.isDefined(
+					String.class, validFieldExpression)) {
+
+				continue;
+			}
+
+			List<Long> userGroupIds = new ArrayList<>();
+
+			UserProcessorContext.UserBind<User> userBind =
+				userProcessorContext.bind(
+					_processingIndex,
+					(currentUser, newUser, serviceContext) -> {
+						_userGroupLocalService.setUserUserGroups(
+							newUser.getUserId(),
+							ArrayUtil.toArray(
+								userGroupIds.toArray(new Long[0])));
+
+						return newUser;
+					});
+
+			userBind.mapStringArray(
+				"userGroups",
+				(user, values) -> {
+					if (values == null) {
+						return;
 					}
-					else if (_log.isWarnEnabled()) {
-						_log.warn("Ignored unknown user group: " + value);
+
+					for (String value : values) {
+						UserGroup userGroup =
+							_userGroupLocalService.fetchUserGroup(
+								user.getCompanyId(), value);
+
+						if (userGroup != null) {
+							userGroupIds.add(userGroup.getUserGroupId());
+						}
+						else if (_log.isWarnEnabled()) {
+							_log.warn("Ignored unknown user group: " + value);
+						}
 					}
-				}
-			});
+				});
+		}
```

Comment:
We should not have a loop here because it will rebind the same userFieldExpression ("userGroups") multiple times if it iterates more than once?